tests/php/bootstrap.php

```diff
@@ -5,6 +5,7 @@
 
 use Yoast\WPTestUtils\WPIntegration;
 
+define( 'WP_DEVELOPMENT_MODE', 'theme' );
```

Comment:
Shouldn't this be `plugin`?

```suggestion
define( 'WP_DEVELOPMENT_MODE', 'plugin' );
```

Reply:
Yes, it should be. I am just testing a few scenarios with `theme`.

Reply:
@westonruter `wp_theme_has_theme_json()` now depends on `wp_get_development_mode()` which is `theme` and `WP_RUN_CORE_TESTS` constant but setting up `WP_RUN_CORE_TESTS` is breaking our multiple test cases so I think for now we have to keep `WP_DEVELOPMENT_MODE` as `theme`.

Reply:
Is the problem that `wp_theme_has_theme_json()` is sending back stale results? Does this reflect a problem with caching?

Reply:
Yes. In these tests, we are changing the theme to `TT3` to test reader theme features from `theme.json`. After changing the theme `wp_theme_has_theme_json()` returns false as the previous theme was not a theme with `theme.json` which is a cached result AFAIK.

Reply:
I think this then reflects a bug in `wp_theme_has_theme_json`. In particular, it is not varying the cache by the theme. There needs to be another static variable in `wp_theme_has_theme_json` which captures the theme that was checked, or else to let `$theme_has_support` be a mapping of theme slug to boolean.

Reply:
Yeah makes sense. It should have some logic to revalidate the cache in case the theme is being changed.

Reply:
Core patch created: https://github.com/WordPress/wordpress-develop/pull/4816